# Employee data analysis

In [60]:
import numpy as np
from numpy import genfromtxt

In [61]:
emp= genfromtxt('DS1_C4_S1_Employee_Data_Challenge.csv',delimiter=',',dtype=None,skip_header=1)

In [62]:
emp

array([[ 1, 22],
       [ 2, 27],
       [ 3, 31],
       [ 4, 29],
       [ 5, 27],
       [ 6, 26],
       [ 7, 27],
       [ 8, 23],
       [ 9, 27],
       [10, 32],
       [11, 28],
       [12, 27],
       [13, 22],
       [14, 27],
       [15, 35],
       [16, 33],
       [17, 28],
       [18, 22],
       [19, 25],
       [20, 27],
       [21, 37],
       [22, 27],
       [23, 24],
       [24, 27],
       [25, 28],
       [26, 26],
       [27, 27],
       [28, 27],
       [29, 29],
       [30, 49],
       [31, 26],
       [32, 32],
       [33, 26],
       [34, 24],
       [35, 24],
       [36, 23],
       [37, 28],
       [38, 28],
       [39, 22],
       [40, 26],
       [41, 32],
       [42, 35],
       [43, 33],
       [44, 31],
       [45, 30],
       [46, 36],
       [47, 22],
       [48, 23],
       [49, 32],
       [50, 34]])

# Task 1

In [63]:
age_column=emp[:,1]

In [64]:
age_column

array([22, 27, 31, 29, 27, 26, 27, 23, 27, 32, 28, 27, 22, 27, 35, 33, 28,
       22, 25, 27, 37, 27, 24, 27, 28, 26, 27, 27, 29, 49, 26, 32, 26, 24,
       24, 23, 28, 28, 22, 26, 32, 35, 33, 31, 30, 36, 22, 23, 32, 34])

In [65]:
avg_age = np.mean(age_column)

In [66]:
avg_age

28.26

Conclusion -  The average of employee is found to be 28.26

# Task 2

In [67]:
age = age_column > 25

In [68]:
age

array([False,  True,  True,  True,  True,  True,  True, False,  True,
        True,  True,  True, False,  True,  True,  True,  True, False,
       False,  True,  True,  True, False,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True, False, False, False,
        True,  True, False,  True,  True,  True,  True,  True,  True,
        True, False, False,  True,  True])

In [69]:
filtered_op= age_column[age]

In [70]:
filtered_op

array([27, 31, 29, 27, 26, 27, 27, 32, 28, 27, 27, 35, 33, 28, 27, 37, 27,
       27, 28, 26, 27, 27, 29, 49, 26, 32, 26, 28, 28, 26, 32, 35, 33, 31,
       30, 36, 32, 34])

In [71]:
count = len(filtered_op)

In [72]:
count

38

Conclusion -The employees who's age is over 25 is]s found and the count is found to be 38 

# Task 3

In [73]:
age1=(age_column>30) &(age_column<35)

In [74]:
age1

array([False, False,  True, False, False, False, False, False, False,
        True, False, False, False, False, False,  True, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False,  True, False, False, False, False,
       False, False, False, False,  True, False,  True,  True, False,
       False, False, False,  True,  True])

In [75]:
filtered=age_column[age1]

In [76]:
filtered

array([31, 32, 33, 32, 32, 33, 31, 32, 34])

In [78]:
new_age_array = np.array([31, 32, 33, 32, 32, 33, 31, 32, 34])

index= np.where(np.isin(emp[:, 1], new_age_array))[0]

matching = emp[index]

print(matching)

[[ 3 31]
 [10 32]
 [16 33]
 [32 32]
 [41 32]
 [43 33]
 [44 31]
 [49 32]
 [50 34]]


Conclusion - The employees in the age group of 30-35 years